packages/db/src/core/cli/migration-queries.ts

```diff
@@ -422,12 +424,46 @@ function hasRuntimeDefault(column: DBColumn): column is DBColumnWithDefault {
 	return !!(column.schema.default && isSerializedSQL(column.schema.default));
 }
 
-export async function getProductionCurrentSnapshot({
-	appToken,
-}: {
+export function getProductionCurrentSnapshot(options: {
 	appToken: string;
 }): Promise<DBSnapshot | undefined> {
-	const url = new URL('/db/schema', getRemoteDatabaseUrl());
+	const remoteUrl = getRemoteDatabaseUrl();
+
+	return isRemoteStudio(remoteUrl)
+		? getStudioCurrentSnapshot(options.appToken, remoteUrl)
+		: getDbCurrentSnapshot(options.appToken, remoteUrl);
+}
+
+async function getDbCurrentSnapshot(
+	appToken: string,
+	remoteUrl: string
+): Promise<DBSnapshot | undefined> {
+	const client = createRemoteDatabaseClient(appToken, remoteUrl);
+
+	try {
+		const res = await client.get<{ snapshot: string }>(
+			// Latest snapshot
+			sql`select snapshot from _astro_db_snapshot order by id desc limit 1;`
+		);
+
+		console.log('DB res:', res);
+
+		return JSON.parse(res.snapshot);
+	} catch (error: any) {
+		if (error.code === 'SQLITE_UNKNOWN') {
+			// Snapshots table was not created yet,
+			return;
```

Comment:
Can you expand the current comment and explain why this error doesn't need to be handled? Still, I think it's valuable to have a log of the error.